docs/openshift-on-azure-walkthrough.md

```diff
@@ -195,11 +199,125 @@ oc login <FQDN> --token=<YOUR_TOKEN>
 ```
 
 
+## AAD application configuration
+
+The deployed OpenShift cluster needs a valid AAD application and service principal to call back into the Azure API,
+in order to enable AAD authentication. There are a few options here:
+
+* Create application using Azure Portal and configure all values there.
+
+* Create application using CLI and later configure permission in the Azure portal.
+
+* Create application using CLI and approve permission in the Azure portal
+
+### Case 1: Create application using Azure portal
+
+Do a search for `App registrations` in the search section located at the tep and navigate to it.
+
+![](./medias/OSA_APP_Portal.png)
+
+Press `New application registration` button. Enter details for your application instance and press `Create`.
+
+![](./medias/OSA_AAD_Portal_Create.png)
+
+ Sign-on URL is OpenShift web console URL: `https://awesome-cluster.eastus.cloudapp.azure.com/oauth2callback/Azure%20AD`
+
+Copy value of the `Application ID` fields for `OSA_AAD_ID` from [step 4 above](#Step-4:-Create-OpenShift-cluster). 
+
+![](./medias/OSA_AAD_Portal_Details.png)
+
+`OSA_AAD_SECRET` Value should be set to `Key`, attached to the AAD application. Under same AAD application press `Keys` and create new key:
```

Comment:
Note: once this key is created and the page is left the value will be hidden.  Make sure to store is somewhere safe.